### PR TITLE
fix: Sessions for Hybrid SDKs

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

The Cocoa SDK didn't start a session after a hybrid app finished its initialization. That's because, in hybrid SDKs, we might initialize the Cocoa SDK after the hybrid engine is ready. The `SessionTracker` then misses the `didBecomeActive` notification from the OS and fails to start a session. To solve this, hybrid SDKs can post an internal notification with the name `SentryHybridSdkDidBecomeActive`. The `SessionTracker` starts a session if it receives such a notification.

Related to https://github.com/getsentry/sentry-dart/pull/274.

## :bulb: Motivation and Context

Fixes GH-912

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the CHANGELOG if needed
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Sources/Sentry/SentrySessionTracker.m [hunk 1]
```diff
@@ -111,10 +130,21 @@ - (void)endCachedSession
 }
 
 /**
- * Is only called when an app is receiving events / it is in the foreground.
+ * Is called when an app is receiving events / it is in the foreground and when we receive a
+ * SentryHybridSdkDidBecomeActiveNotification. There is no guarantee if this method is called once
+ * or twice. We need to ensure that we execute it only once.
```
Comment (marandaneto) [inline Sources/Sentry/SentrySessionTracker.m:135]:
```suggestion
 * It is called when an App. is receiving events / It is in the foreground and when we receive a
 * SentryHybridSdkDidBecomeActiveNotification. There is no guarantee that this method is called once
 * or twice. We need to ensure that we execute it only once.
```

File: Sources/Sentry/SentrySessionTracker.m [hunk 2]
```diff
@@ -111,10 +130,21 @@ - (void)endCachedSession
...
  */
 - (void)didBecomeActive
 {
+    // We don't know if the hybrid SDKs post the notification from a background thread, so we
+    // synchronize to be safe.
```
Comment (marandaneto) [inline Sources/Sentry/SentrySessionTracker.m:140]: the sync. is needed, but the comment is off for me, it's not really about a background thread, it's about receiving 2 notifications at the same time and racing each other

**Review by marandaneto — APPROVED** [review]

File: Sources/Sentry/SentrySessionTracker.m (hunk 2 above)
Comment (philipphofmann) [reply Sources/Sentry/SentrySessionTracker.m:140]: If we would receive both notifications on the main thread, they can't race. Usually, you receive the `didBecomeActive` notification on the main thread, so we don't have any synchronization right until now. The callback is executed on the thread you post the notification. As I don't know on which thread the hybrid SDKs might post the notification I added the synchronize.

Comment (marandaneto) [reply Sources/Sentry/SentrySessionTracker.m:140]: I do agree with you, but what I mean is, I'm not entirely sure that the main thread on Android OS is actually the very same main thread on Flutter, as they use the Message channel for these things, still, it'd be considered the main thread.
reading again your comment, it does make sense.

**Pull request merged**